lib/grant-types/abstract-grant-type.js

```diff
@@ -0,0 +1,95 @@
+
```

Comment:
There doesn't seem to be a validation of the client in any of the grant types.  It is common practice to require authenticating the client (with client and secret) when processing the grant and generating a token.